spring-shell2-shell1-adapter/src/main/java/org/springframework/shell2/legacy/LegacyParameterResolver.java

```diff
@@ -82,7 +86,40 @@ public Object resolve(MethodParameter methodParameter, List<String> words) {
 
 	@Override
 	public ParameterDescription describe(MethodParameter parameter) {
-		throw new UnsupportedOperationException();
+		Parameter jlrParameter = parameter.getMethod().getParameters()[parameter.getParameterIndex()];
+		CliOption option = jlrParameter.getAnnotation(CliOption.class);
+		ParameterDescription result = ParameterDescription.outOf(parameter);
+		result.help(option.help());
+		result.defaultValue(defaultValueFor(option));
+		result.keys(Arrays.asList(option.key()));
+		result.mandatoryKey(false);
```

Comment:
You are correct in your understanding. Which means you're wrong in your code I think :) 
It should say `result.mandatoryKey(true)` (because in shell 1, you **need** to specify the `--optionName` prefix (unless it's the "default" option))

See my general comment for this PR